tests/test_plain_html_functions.py

```diff
@@ -29,3 +29,28 @@ def test_remove_blacklist():
     soup = BeautifulSoup(html, "html5lib")
     plain_html.remove_blacklist(soup)
     assert "button" not in str(soup)
+
+
+def test_remove_cdata():
```

Comment:
If I understand this refactor correctly, we are testing that CData is removed as expected even though in all cases it is removed as a by product of some other processing (parsing or Script removal). This is a good safety net that ensures our tests break if we switch to a non-CData-stripping parser or retain a text representation of script tag content (less likely).